volatility3/framework/plugins/windows/idt.py

```diff
@@ -0,0 +1,295 @@
+from typing import List
+
+from volatility3.framework import interfaces
+from volatility3.framework.renderers import TreeGrid, format_hints
+from volatility3.framework.interfaces import plugins
+from volatility3.framework.configuration import requirements
+from volatility3.plugins.windows import modules
+
+GDT_DESCRIPTORS = dict(
+    enumerate(
+        [
+            "Data RO",
+            "Data RO Ac",
+            "Data RW",
+            "Data RW Ac",
+            "Data RO E",
+            "Data RO EA",
+            "Data RW E",
+            "Data RW EA",
+            "Code EO",
+            "Code EO Ac",
+            "Code RE",
+            "Code RE Ac",
+            "Code EO C",
+            "Code EO CA",
+            "Code RE C",
+            "Code RE CA",
+            "<Reserved>",
+            "TSS16 Avl",
+            "LDT",
+            "TSS16 Busy",
+            "CallGate16",
+            "TaskGate",
+            "Int Gate16",
+            "TrapGate16",
+            "<Reserved>",
+            "TSS32 Avl",
+            "<Reserved>",
+            "TSS32 Busy",
+            "CallGate32",
+            "<Reserved>",
+            "Int Gate32",
+            "TrapGate32",
+        ]
+    )
+)
+
+
+class _KIDT:
+    def __init__(self, idt_struct):
+        self.idt = idt_struct
+        self.Offset = idt_struct.Offset
+        self.Selector = idt_struct.Selector
+        self.Access = idt_struct.Access
+        self.ExtendedOffset = idt_struct.ExtendedOffset
+
+    @property
+    def Address(self):
+        if self.ExtendedOffset:
+            return self.ExtendedOffset << 16 | self.Offset
+
+        return 0
+
+
+class _KGDT:
+    def __init__(self, gdt_struct):
+        self.gdt = gdt_struct
+        self.LimitLow = gdt_struct.LimitLow
+        self.BaseLow = gdt_struct.BaseLow
+        self.HighWord = gdt_struct.HighWord
+
+    @property
+    def Type(self):
+        """Get a string name of the descriptor type"""
+        flag = self.HighWord.Bits.Type & 1 << 4
+        typeval = self.HighWord.Bits.Type & ~(1 << 4)
+
+        if flag == 0:
+            typeval += 16
+
+        return GDT_DESCRIPTORS.get(typeval, "UNKNOWN")
+
+    @property
+    def Base(self):
+        """Get the base (start) of memory for this GDT"""
+        return self.BaseLow + (
+            (self.HighWord.Bits.BaseMid + (self.HighWord.Bits.BaseHi << 8)) << 16
+        )
+
+
+class _KPCR:
+    def __init__(self, kpcr_obj, ntkrnlmp, layer_name, symbol_table):
+        self.kpcr = kpcr_obj
+        self.ntkrnlmp = ntkrnlmp
+        self.layer_name = layer_name
+        self.symbol_table = symbol_table
+
+    def idt_entries(self):
+        base_idt = self.kpcr.IDT
```

Comment:
I believe this may only work for x86. On x64, the `_KPCR` finds its IDT through a member named `IdtBase` instead. That may be OK if you only plan to support 32-bit systems, however the plugins' list of architectures include `Intel64`.